src/core/column/sumprod.h

```diff
@@ -31,55 +31,51 @@ namespace dt {
   class SumProd_ColumnImpl : public Virtual_ColumnImpl {
   private:
     Column col_;
-    bool is_grouped_;
     Groupby gby_;
+    bool is_grouped_;
+    size_t : 56;
 
   public:
-    SumProd_ColumnImpl(Column &&col, bool is_grouped, const Groupby &gby)
-      :Virtual_ColumnImpl(gby.size(), col.stype()),
-       col_(std::move(col)),
-       is_grouped_(is_grouped),
-       gby_(gby)
+    SumProd_ColumnImpl(Column &&col, Groupby &&gby, bool is_grouped)
```

Comment:
@oleksiyskononenko what's the idea behind const vs && in this case?

Reply:
If you look at the way we are getting the groupby object from `ctx`, i.e.
```python
Groupby gby = ctx.get_groupby();
```
you will notice, that we essentially create a copy of that object. Unfortunately, we can't do
```python
const Groupby& gby = ctx.get_groupby();
```
to avoid copying, because later we may need to assign a single group to `gby`.

Actually, creating a copy is ok in this case, because we need to store that groupby object safely in `SumProd_ColumnImpl`. However, we are passing this copy around as `const Groupby& gby`, so when it comes to `SumProd_ColumnImpl` we have to create its own copy of the object. To prevent this additional copy from being created, it would be nice to just `std::move()` the groupby object around. 

I found that `fexpr_sumprod.cc` is not the only place where we may need to make this change, so it would be nice to have a separate PR with the fix.

Reply:
Also, when we do reduction for several columns at the same time, we do need as many groupby copies as the number of the columns, so we can't simply `std::move()` the initial groupby. Anyways, I've reverted these changes as this issue needs more understanding.